robottelo/hosts.py

```diff
@@ -164,19 +164,36 @@ def satellite(self):
     @property
     def nailgun_host(self):
         """If this host is subscribed, provide access to its nailgun object"""
-        if self.subscribed:
+        if self.identity.get('registered_to') == self.satellite.hostname:
```

Comment:
It is likely to fail if the host is registered to a capsule

Reply:
this is simply referring to the optional `satellite` attribute of the `ContentHost` class.
https://github.com/SatelliteQE/robottelo/blob/master/robottelo/hosts.py#L144
there doesn't seem to be any distinction made between satellite and capsule ATM.

I believe this is as closest as we can get to the real state as we can with the current implementation

Reply:
I am going to test this scenario using the code from @rplevka's branch next week. I will let you know the result.

Reply:
If a ContentHost is registered to a capsule, it shows up in the host list in Satellite, so we should be able to find it there.
If a Capsule is registered to Satellite, then `nailgun_host` for the capsule should exist too, but the question is whether we can determine the correct Satellite instance to search the Host entity in.
Answer: We surely can if Capsule's `satellite` property gives us the right Satellite instance.

@Gauravtalreja1 Could you correct me if I am wrong here?

Reply:
After some testing, I see that if the host is registered to Capsule, `rhsm.conf` contains the Capsule hostname, not the Satellite one. #10547 will not save us, the condition needs to be tweaked.